app/src/main/java/com/realityexpander/tasky/agenda_feature/data/repositories/agendaRepository/agendaRepositoryImpls/AgendaRepositoryImpl.kt

```diff
@@ -65,87 +74,106 @@ class AgendaRepositoryImpl @Inject constructor(
         }
     }
 
-    override suspend fun updateLocalAgendaDayFromRemote(dateTime: ZonedDateTime) {
-        withContext(Dispatchers.IO) {
-            try {
-                // Get fresh data
-                val result = agendaApi.getAgenda(dateTime)
-
-                if (result.isSuccess) {
-                    val agenda = result.getOrNull()
-
-                    eventRepository.clearEventsForDayLocally(dateTime)
-                    // Insert fresh data locally
-                    agenda?.events?.forEach { event ->
-                        val result2 =
-                            eventRepository.upsertEventLocally(event.toDomain())
-                        if (result2 is ResultUiText.Error) {
-                            throw IllegalStateException(result2.message.asStrOrNull())
-                        }
+    override suspend fun updateLocalAgendaDayFromRemote(dateTime: ZonedDateTime): ResultUiText<Unit> {
+        try {
+            // Get fresh data
+            val result = agendaApi.getAgenda(dateTime)
+
+            if (result.isSuccess) {
+                val agenda = result.getOrNull()
+
+                eventRepository.clearEventsForDayLocally(dateTime)
+                // Insert fresh data locally
+                agenda?.events?.forEach { event ->
+                    val result2 =
```

Comment:
Usually I'd avoid using trailing numbers for variable names since it's not very expressive, I'd stick to `getAgendaResult` and `upsertEventResult` here.